samples/ConPTY/EchoCon/EchoCon/EchoCon.cpp

```diff
@@ -0,0 +1,197 @@
+// EchoCon.cpp : Entry point for the EchoCon sample application.
+//
+
+#include "stdafx.h"
+#include <Windows.h>
+#include <conio.h>
+#include <process.h>
+#include <wchar.h>
+
+// Forward declarations
+void __cdecl PipeListener(LPVOID);
+HRESULT InitializeStartupInfoAttachedToPseudoConsole(STARTUPINFOEX*, HPCON);
+COORD GetConsoleSize(HANDLE);
+
+int main()
+{
+    wchar_t szCommand[] = L"ping localhost";
+    HRESULT hr{ S_OK };
+    HANDLE hPipeIn{ INVALID_HANDLE_VALUE };
+    HANDLE hPipeOut{ INVALID_HANDLE_VALUE };
+    HANDLE hPipePTYIn{ INVALID_HANDLE_VALUE };
+    HANDLE hPipePTYOut{ INVALID_HANDLE_VALUE };
+    HPCON hPC{ INVALID_HANDLE_VALUE };
+
+    PROCESS_INFORMATION piClient{};
+    STARTUPINFOEX startupInfo{};
+
+    HANDLE hConsole = { GetStdHandle(STD_OUTPUT_HANDLE) };
+    DWORD consoleMode{};
+
+    // Enable VT Processing
+    GetConsoleMode(hConsole, &consoleMode);
+    hr = SetConsoleMode(hConsole, consoleMode | ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+        ? S_OK
+        : GetLastError();
+    if (S_OK == hr)
+    {
+        // Create the pipes to which the ConPTY will connect
+        if (CreatePipe(&hPipePTYIn, &hPipeOut, NULL, 0) &&
+            CreatePipe(&hPipeIn, &hPipePTYOut, NULL, 0))
+        {
+            // Create & start thread to listen to the incoming pipe
+            // NB: Using CRT-safe _beginthread() rather than CreateThread()
+            DWORD pipeListenerThreadId{};
+            HANDLE hPipeListenerThread{ 
+                reinterpret_cast<HANDLE>(_beginthread(PipeListener, 0, hPipeIn)) };
+
+            // Create appropriately sized ConPTY
+            COORD consoleSize{ GetConsoleSize(hConsole) };
+            hr = CreatePseudoConsole(consoleSize, hPipePTYIn, hPipePTYOut, 0, &hPC);
+
+            // Initialize the necessary startup info struct        
+            if (S_OK == InitializeStartupInfoAttachedToPseudoConsole(&startupInfo, hPC))
+            {
+                // Launch ping to echo some text back via the pipe
+                hr = CreateProcess(
+                    NULL,                           // No module name - use Command Line
+                    szCommand,                      // Command Line
+                    NULL,                           // Process handle not inheritable
+                    NULL,                           // Thread handle not inheritable
+                    TRUE,                           // Inherit handles
```

Comment:
I believe it's safer to pass FALSE for this param.

I thought we fixed whatever the issue was with the client app inheriting the handles of the console application acting as a terminal, but for whatever reason node-pty was still acting weird if we passed TRUE here. 

I think it'd be safer to have this param as FALSE, so no-one else hits this gotcha, at least until we figure out exactly what's wrong with node-pty.

Reply:
Thanks. Fixed.